packages/snaps-execution-environments/src/common/endowments/index.ts

```diff
@@ -23,21 +18,19 @@ type EndowmentFactoryResult = {
   [key: string]: unknown;
 };
 
+/**
+ * Retrieve consolidated endowment factories for common endowments.
+ */
+const registeredEndowments = buildCommonEndowments();
+
 /**
  * A map of endowment names to their factory functions. Some endowments share
  * the same factory function, but we only call each factory once for each snap.
  * See {@link createEndowments} for details.
  */
-const endowmentFactories = [
-  timeout,
-  interval,
-  network,
-  crypto,
-  math,
-  date,
-].reduce((factories, builder) => {
+const endowmentFactories = registeredEndowments.reduce((factories, builder) => {
   builder.names.forEach((name) => {
-    factories.set(name, builder.factory);
+    factories.set(name, builder.factory as () => EndowmentFactoryResult);
```

Comment:
Since I made a change to have factory for endowment factories, it started detecting this as an issue:
```
TS2345: Argument of type '() => unknown' is not assignable to parameter of type '() => EndowmentFactoryResult'.   Type 'unknown' is not assignable to type 'EndowmentFactoryResult'.
```

Because the starting value is `new Map<string, () => EndowmentFactoryResult>()` and `EndowmentFactoryResult` type (defined above) is not completely matching it.

Reply:
Let me know if you have a better solution to this. I can try to implement.

Reply:
I've updated endowment factory type so this is no longer needed.